tests/tests_insights_ansible_host.yaml

```diff
@@ -0,0 +1,116 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Insights ansible host test
+  hosts: all
+  gather_facts: false
+  become: true
+
+  tasks:
+    - name: Setup Insights
+      import_tasks: tasks/setup_insights.yml
+
+    - name: Test ansible_host
+      block:
+        - name: Add ansible_host and register insights
+          include_role:
+            name: linux-system-roles.rhc
+            public: true
+          vars:
+            rhc_auth:
+              login:
+                username: "{{ lsr_rhc_test_data.reg_username }}"
+                password: "{{ lsr_rhc_test_data.reg_password }}"
+            rhc_insights:
+              ansible_host: host
+              remediation: absent
+              state: present
+            rhc_organization: "{{ lsr_rhc_test_data.reg_organization }}"
+            rhc_server:
+              hostname: "{{ lsr_rhc_test_data.candlepin_host }}"
+              port: "{{ lsr_rhc_test_data.candlepin_port }}"
+              prefix: "{{ lsr_rhc_test_data.candlepin_prefix }}"
+              insecure: "{{ lsr_rhc_test_data.candlepin_insecure }}"
+            rhc_baseurl: "{{ lsr_rhc_test_data.baseurl | d(omit) }}"
+
+        - name: Check ansible_host is set to 'host' in config file
+          command:
+            grep -ixq "^ansible_host=host" {{ __rhc_insights_conf }}
+          changed_when: true
+
+        - name: Test ansible_host changed value after registration
+          block:
+            - name: Change ansible host to 'new-host'
+              include_role:
```

Comment:
> I wonder. Since the master task “Test ansible host” includes the role and sets _rhc_insights.state_ to _present_, it registers the system at its very start. This block, including the role again, triggers the registration on an already registered host?

No, because the role is smart enough to see that the system is already registered.  One important concept in Ansible is *idempotency* - running the role again with the same arguments should not change the state of the system, and Ansible should report no `"changed": true`.

> 
> If so, I’m afraid that the test scenario can be possibly affected by the tainted state.

I don't think so.

> In such case it would be necessary to have the test in its own playbook, which I think is generally a good idea when testing an Ansible role.

But not in this case.

> 
> In this case, the test should manually edit the config file by an _lineinfile_ task, then trigger the registration, and after that check that the value in the config file changed.

I don't think that's necessary in this case.

Reply:
I see. The first task does the registration, and then, this one test that `ansible_host` is updated on an already registered host. Ok!